src/main/daml/Daml/Finance/Claims/Util/Builders.daml

```diff
@@ -131,20 +131,14 @@ createAssetPerformancePaymentClaims dateToTime schedule periodicSchedule useAdju
   ownerReceives dayCountConvention notional cashInstrumentCid referenceAssetId =
   let
     assetClaimAmounts = mconcat $ map (\p ->
-        when (TimeGte p.adjustedStartDate)
-        $ scale (Const 1.0 / Observe referenceAssetId)
-        $ when (TimeGte p.adjustedEndDate)
-        $ scale (Observe referenceAssetId)
-        $ scale (Const notional)
+      let performance = Observe referenceAssetId / ObserveAt referenceAssetId p.adjustedStartDate - Const 1.0
```

Comment:
I think this is definitely a more user-friendly way of writing these kind of expressions.

Reply:
An equivalent (and probably even more user-friendly way) of writing the payoff would be
`let performance = ObserveAt referenceAssetId p.adjustedEndDate / ObserveAt referenceAssetId p.adjustedEndDate`